src/clojure/clj_yaml/core.clj

```diff
@@ -156,3 +157,18 @@
                             :allow-recursive-keys allow-recursive-keys
                             :allow-duplicate-keys allow-duplicate-keys)
                  string) keywords))
+
+;; From https://github.com/metosin/muuntaja/pull/94/files
+(defn generate-stream
+  "Dump the content of data as yaml into writer."
+  [writer data & opts]
+  (.dump ^Yaml (apply make-yaml opts) (encode data) writer))
+
+(defn parse-stream
+  [^InputStream data & {:keys [unsafe mark keywords max-aliases-for-collections allow-recursive-keys allow-duplicate-keys] :or {keywords true}}]
```

Comment:
Why take in InputStream here and not a reader - the first argument in generate-stream is a writer.